sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -374,6 +374,29 @@ static std::string getUint32PropAsOptStr(const RTDeviceBinaryImage &Img,
   return temp;
 }
 
+static void appendCompileOptionsForRegAllocMode(std::string &CompileOpts,
+                                                const RTDeviceBinaryImage &Img,
+                                                bool IsEsimdImage) {
+  pi_device_binary_property Prop = Img.getProperty("RegisterAllocMode");
+  if (!Prop)
+    return;
+  uint32_t PropVal = DeviceBinaryProperty(Prop).asUint32();
+  // 2 means Large GRF.
+  if (PropVal == 2) {
+    if (!CompileOpts.empty())
+      CompileOpts += " ";
+    CompileOpts += IsEsimdImage ? "-doubleGRF" : "-ze-opt-large-register-file";
```

Comment:
Will `-ze-opt-large-register-file` also work on OpenCL?

Reply:
yes, i will add a comment, this is existing code tho. thx